sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/QuantileSummaries.scala

```diff
@@ -236,7 +240,7 @@ object QuantileSummaries {
     if (currentSamples.isEmpty) {
       return Array.empty[Stats]
     }
-    val res: ArrayBuffer[Stats] = ArrayBuffer.empty
+    val res = ListBuffer.empty[Stats]
```

Comment:
Ah, thanks, I thought it was smart enough to use a circular buffer, thanks! It would be nice to have more control over the memory eventually, but this is fine for now.